provider/vsphere/internal/vsphereclient/createvm.go

```diff
@@ -170,20 +170,30 @@ func (c *Client) ensureTemplateVM(
 	if err != nil {
 		return nil, errors.Trace(err)
 	}
-	templateVM, err := finder.VirtualMachine(ctx, path.Join(vmTemplatePath(args), vmTemplateName(args)))
-	if err == nil && templateVM != nil {
-		return templateVM, nil
+	// Finding the folder can be tricky since user may pass same
```

Comment:
the `arg.Folder` contains `vmFolderAttr` which could be relative path or full path.
Removing the comment for less confusion since everything handled inside `client.FindFolder` already.